### PR TITLE
Docker image with dockerfile

### PR DESCRIPTION
**Pull request by eventh**
Ideally. the installation of dependencies should happen before we copy over the code. But poetry does something strange with the package command in the pyproject.toml file, where cognite-replicator is installed as a package in the venv. This then leads to it being a dependency in the lock-file, which again means you need the code there before you install dependencies in the build step in the dockerfile..

Anyway, still works.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  PIP_DISABLE_PIP_VERSION_CHECK=on \
+  PIP_DEFAULT_TIMEOUT=100 \
+  POETRY_VERSION=0.12.16
+
+# System deps:
+RUN pip install "poetry==$POETRY_VERSION"
+
+# Copy only requirements to cache them in docker layer
+WORKDIR /code
+COPY poetry.lock pyproject.toml /code/
```
Comment (tristan-cognite) [inline Dockerfile]: Seems redundant with line 21 - both copy these files to /code

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+# Creating folders, and files for a project:
+COPY . /code
+
+# Project initialization:
+RUN poetry config settings.virtualenvs.create false && poetry install --no-dev --no-interaction --no-ansi
+
+ENTRYPOINT ["python", "-m", "cognite.replicator"]
+CMD []
```
Comment (tristan-cognite) [inline Dockerfile]: I'm not sure if we need this line.

Comment (eventh) [reply Dockerfile]: I'm assuming that CMD is inherited from the parent dockerfile [1] (not sure if it is true).
See [2] for discussion on ENTRYPOINT vs CMD. I think this setup is close to best practice.

[1] https://github.com/docker-library/python/blob/c3233a936f58bee7c6899d3e381f23ed12cfc7a8/3.7/buster/Dockerfile
[2] https://aws.amazon.com/blogs/opensource/demystifying-entrypoint-cmd-docker/

Comment (eventh) [reply Dockerfile]: I tested the assumption, and it was not needed. Removed the CMD. Good suggestion! :)

**Pull request merged**